lib/postgrex.ex

```diff
@@ -66,6 +66,13 @@ defmodule Postgrex do
     if not `DBConnection.Connection` (default: `DBConnection.Connection`);
     * `:null` - The atom to use as a stand in for postgres' `NULL` in encoding
     and decoding (default: `nil`);
+    * `:date` - Either `:postgrex` to use Postgrex date structs or
```

Comment:
You are right. It's a good idea to do the backwards incompatible change before 1.0 but still keep a compatible alternative as an option.